internal/openshift/controller.go

```diff
@@ -145,9 +145,10 @@ func (c *controllerImpl) HandleDeploymentConfig(dc model.DCObject) error {
 	user := userIdler.GetUser()
 	sendUserToIdler := false
 
-	condition, err := dc.Object.Status.GetByType(availableCond)
-	if err != nil {
-		return err
+	condition := dc.Object.Status.GetByType(availableCond)
+	if condition.Type == "" {
```

Comment:
I don't see why condition.Type would come back with empty string according to the code :

https://github.com/fabric8-services/fabric8-jenkins-idler/blob/e501e24a559a261eb48afd3aab48364dd3690a33/internal/model/openshift_objects.go#L189-L198

it should come back with an error if it's not matching, (and due the previous code would catch it)

Reply:
@kishansagathiya @chmouel How about 

```patch
diff --git a/internal/openshift/controller.go b/internal/openshift/controller.go
index d82d244..8371045 100644
--- a/internal/openshift/controller.go
+++ b/internal/openshift/controller.go
@@ -147,7 +147,9 @@ func (c *controllerImpl) HandleDeploymentConfig(dc model.DCObject) error {
 
 	condition, err := dc.Object.Status.GetByType(availableCond)
 	if err != nil {
-		return err
+		log.Infof("Available condition not present (%s) in the list of conditions - SKIPPING", err)
+		// stop processing since the pod isn't available yet
+		return nil
 	}
 
 	// TODO Verify if we need Generation vs. ObservedGeneration

```